### PR TITLE
Add new node select-knx-groupaddress for comfortable selection of knx…

### PR DESCRIPTION
**Pull request by FloMaetschke**
… group addresses by a csv export of ETS.
This is a quick hack to simplify access of your group addresses in node red.

### CODE REVIEW

**Comment by ekarak** [comment]
Apologies for not reviewing this earlier, but I'm in the process of rewriting a *proper* KNX access library, and get rid of all the eibd stuff. See https://www.npmjs.com/package/knx for more details. 

I'll probably review and merge your effort as I see its very useful. Is there any way we can access ETS database directly without the need to export a CSV file? (ie. using a direct db connection?)

**Comment by FloMaetschke** [comment]
Hey!

No problem. I am sorry. It seems to be the fastest and easiest way to do. I dont know any xml parser for knx ets projects. There is also a Problem with the ets versions 3,4 and 5 having different formats.
Csv seems to be the sinplest way and can also be made by hand.

Greetings!
Flo


Von meinem Samsung Galaxy Smartphone gesendet.


-------- Ursprüngliche Nachricht --------
Von: Elias Karakoulakis <notifications@github.com>
Datum: 24.09.16 06:56 (GMT+01:00)
An: ekarak/node-red-contrib-eibd <node-red-contrib-eibd@noreply.github.com>
Cc: Florian Mätschke <florian@maetschke.com>, Author <author@noreply.github.com>
Betreff: Re: [ekarak/node-red-contrib-eibd] Add new node select-knx-groupaddress for comfortable selection of knx… (#9)


Apologies for not reviewing this earlier, but I'm in the process of rewriting a proper KNX access library, and get rid of all the eibd stuff. See https://www.npmjs.com/package/knx for more details.

I'll probably review and merge your effort as I see its very useful. Is there any way we can access ETS database directly without the need to export a CSV file? (ie. using a direct db connection?)

**Comment by FloMaetschke** [comment]
Hey your package looks nice!
I was using eibd the last day and have given up on reading the correct response value. It always 0 but not 1 if the lamp is on.

I will try to use the knx api now. Lets see how far i come ;)

Thx so much,
Florian


Von: Elias Karakoulakis [mailto:notifications@github.com]
Gesendet: Samstag, 24. September 2016 06:57
An: ekarak/node-red-contrib-eibd <node-red-contrib-eibd@noreply.github.com>
Cc: Florian Mätschke <florian@maetschke.com>; Author <author@noreply.github.com>
Betreff: Re: [ekarak/node-red-contrib-eibd] Add new node select-knx-groupaddress for comfortable selection of knx… (#9)


Apologies for not reviewing this earlier, but I'm in the process of rewriting a proper KNX access library, and get rid of all the eibd stuff. See https://www.npmjs.com/package/knx for more details.

I'll probably review and merge your effort as I see its very useful. Is there any way we can access ETS database directly without the need to export a CSV file? (ie. using a direct db connection?)

File: select-knx-groupaddress/select-knx-groupaddress.html [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+
+ oneditprepare: function() {
+        var previous = null;
+
+	    // get the config node using the ID:
+	    var configNode = RED.nodes.node(this.groupaddresses);
+
+        // parse csv and generate a groupaddresses array:
+        var csv = processCsv(configNode.csv);
```
Comment (ekarak) [inline select-knx-groupaddress/select-knx-groupaddress.html]: this causes an exception when no config node exists

**Comment by ekarak** [comment]
Hi Florian,
I will copy your PR along with necessary changes over to https://bitbucket.org/ekarak/node-red-contrib-knxjs
Thanks

**Comment by FloMaetschke** [comment]
Nice! thx

Von: Elias Karakoulakis [mailto:notifications@github.com]
Gesendet: Sonntag, 9. Oktober 2016 18:24
An: ekarak/node-red-contrib-eibd <node-red-contrib-eibd@noreply.github.com>
Cc: Florian Mätschke <florian@maetschke.com>; Author <author@noreply.github.com>
Betreff: Re: [ekarak/node-red-contrib-eibd] Add new node select-knx-groupaddress for comfortable selection of knx… (#9)


Hi Florian,
I will copy your PR along with necessary changes over to https://bitbucket.org/ekarak/node-red-contrib-knxjs
Thanks

**Pull request closed without merging**